hb.ml

```diff
@@ -0,0 +1,194 @@
+
+(* ----------- aux -------------------------------------------------------- *)
+let int_of_string s =
+  try int_of_string s
+  with Failure _ -> Printf.eprintf "ERROR: not a number \n"; exit 1
+
+let unix_read ic len =
+  let all = Bytes.create len in
+  let rec loop cur len =
+    if cur >= len then ()
+    else
+      let n = Unix.read ic all cur (len-cur) in
+      loop (cur+n) len in
+  loop 0 len;
+  Bytes.to_string all
+
+let unix_open_file file =
+  let ic = Unix.openfile file [Unix.O_RDONLY] 0 in
+  let len = Unix.lseek ic 0 Unix.SEEK_END in
+  let _ = Unix.lseek ic 0 Unix.SEEK_SET in
+  ic, len
+
+let unix_read_whole_file file =
+  let ic, len = unix_open_file file in
+  let data = unix_read ic len in
+  Unix.close ic;
+  data
+
+(* ----------- patches ----------------------------------------------------- *)
```

Comment:
Did you invent a new patch format? Can't you generate a unix diff by comparing the original file with the expansion of hb commands (or directly generate something in diff syntax) and use standard tools?

Reply:
I can't generate patches in the format we like (the unified one, the one with ++ and --) because I don't have the context, only the loc. There are other diff formats, but I don't see the point. I mean `hb patch foo.v; git diff` does what you want.